### PR TITLE
Added Baremetal-Blinky example

### PR DESCRIPTION
**Pull request by linlingao**

### CODE REVIEW

**Comment by linlingao** [comment]
@bridadan you're the proud reviewer chosen by Amanda.

File: Baremetal-Blinky/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## Baremetal Blinky example using Mbed OS
+
+This guide reviews the steps required to build and run a baremetal Blinky application on Ti's CC3220SF LaunchXL platform without using DigitalOut class and timers.
+
+Please install [Mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
+
+This example retrieves code from https://github.com/ARMmbed/mbed-os-ti-port. It's a private repo, please contact support@mbed.com to request access.
```
Comment (bridadan) [inline Baremetal-Blinky/README.md]: Any reason to keep this repository private? Seems counter intuitive for an example!

File: Baremetal-Blinky/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+### Import the example application
+
+From the command-line, import the example:
+
+```
+git clone https://github.com/ARMmbed/mbed-os-examples-docs_only.git
+cd Baremetal-Blinky
+mbed config root .
+mbed deploy
```
Comment (bridadan) [inline Baremetal-Blinky/README.md]: You can shorten this to the following:

```
mbed import https://github.com/ARMmbed/mbed-os-examples-docs_only
cd Baremetal-Blinky
```

`mbed import` takes care of cloning, `mbed config root .`, and `mbed deploy`.

**Comment by kegilbert** [comment]
Wow no kidding, that was really bare metal! Really cool example to have. This was already merged so sorry for continuing conversation but just curious since it might be cool to add; how big is the binary compared to traditional blinky with the full OS? I'd imagine this is quite a bit smaller.

File: Baremetal-Blinky/README.md (hunk 2 above)
Comment (cmonr) [reply Baremetal-Blinky/README.md]: Oh boy. Instructions are starting to be made using what was originally meant to be a temporary repo.

I think I would have prefered the README.md be made asif the subdirectory were its own repo.

File: Baremetal-Blinky/mbed-os.lib [hunk 3]
```diff
@@ -0,0 +1 @@
+https://github.com/ARMmbed/mbed-os-ti-port/#1304ebdb5e8d1e6d13aabe1360813b75f58c0828
```
Comment (cmonr) [inline Baremetal-Blinky/mbed-os.lib]: Woah, interesitng.

Will be nice to see this eventually updated to ARMmbed/mbed-os one day 😄

File: Baremetal-Blinky/README.md (hunk 1 above)
Comment (cmonr) [reply Baremetal-Blinky/README.md]: Think it might have to do with where this doc is being referenced to begin with? 

@AnotherButler Where is this example living in the docs anywho?
